extensions/amp-social-share/0.2/storybook/Basic.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {SocialShare} from '../social-share';
+import {addParamsToUrl} from '../../../../src/url';
+import {getSocialConfig} from '../amp-social-share-config';
+import {select, text, withKnobs} from '@storybook/addon-knobs';
+import {withA11y} from '@storybook/addon-a11y';
+
+export default {
+  title: 'Social Share',
+  component: SocialShare,
+  decorators: [withA11y, withKnobs],
+};
+
+export const _default = () => {
+  const knobConfigurations = [
+    'email',
+    'facebook',
+    'linkedin',
+    'pinterest',
+    'tumblr',
+    'twitter',
+    'whatsapp',
+    'line',
+    'sms',
+    'system',
+    'custom endpoint',
```

Comment:
Is it possible to create an input for `type` that's outside of the preconfigured providers? i.e. a select dropdown of preconfigured providers or an empty input to configure the `type` prop value, since really publishers can put anything they want, not just `"custom endpoint"`.

Reply:
It is possible, but any 'type' attribute that is not one of the pre-configured types, actually does not change the behavior of the component.  So having the 'custom endpoint' as a catch all should address this use case.  Unless there is something else I'm mising.

Reply:
@caroqliu could you clarify? It also looks to me that a single "custom endpoint" should suffice. Do you see the need for more?